tests/visualization_tests/test_optimization_history.py

```diff
@@ -209,11 +221,57 @@ def objective(trial: Trial) -> float:
     for x, study in zip(suggested_params, studies):
         study.enqueue_trial({"x": x})
         study.optimize(objective, n_trials=1)
-    figure = plot_optimization_history(studies, error_bar=True)
-
-    mean = np.mean(suggested_params)
-    std = np.std(suggested_params)
-
-    np.testing.assert_almost_equal(figure.data[0].y, mean)
-    np.testing.assert_almost_equal(figure.data[2].y, mean + std)
-    np.testing.assert_almost_equal(figure.data[3].y, mean - std)
+    info_list = _get_optimization_history_info_list(
+        studies, target=None, target_name="Objective Value", error_bar=True
+    )
+    mean = np.mean(suggested_params).item()
+    std = np.std(suggested_params).item()
+    assert info_list == [
+        _OptimizationHistoryInfo(
+            [0],
+            _ValuesInfo([mean], [std], "Objective Value"),
+            _ValuesInfo([mean], [std], "Best Value"),
+        )
+    ]
+
+
+@pytest.mark.parametrize("target_name", ["Objective Value", "Target Name"])
+@pytest.mark.parametrize(
+    "info_list",
+    [
+        [],  # Empty info.
+        [  # Vanilla.
+            _OptimizationHistoryInfo(
+                [0, 1, 2],
+                _ValuesInfo([1.0, 2.0, 0.0], None, "Dummy"),
+                None,
+            )
+        ],
+        [  # With best values.
+            _OptimizationHistoryInfo(
+                [0, 1, 2],
+                _ValuesInfo([1.0, 2.0, 0.0], None, "Dummy"),
+                _ValuesInfo([1.0, 1.0, 1.0], None, "Best Value"),
+            )
+        ],
+        [  # With error bar.
+            _OptimizationHistoryInfo(
+                [0, 1, 2],
+                _ValuesInfo([1.0, 2.0, 0.0], [1.0, 2.0, 0.0], "Dummy"),
+                None,
+            )
+        ],
+        [  # With best values and error bar.
+            _OptimizationHistoryInfo(
+                [0, 1, 2],
+                _ValuesInfo([1.0, 2.0, 0.0], [1.0, 2.0, 0.0], "Dummy"),
+                _ValuesInfo([1.0, 1.0, 1.0], [1.0, 2.0, 0.0], "Best Value"),
+            )
+        ],
+    ],
+)
+def test_get_optimization_history_plot(
```

Comment:
How about parametrizing `_get_optimization_history_plot` with `matplotlib`?
You can still branch the assertion based on the type of the returned figure.

Reply:
What about https://github.com/optuna/optuna/pull/3806/commits/7a3f807b48abf76b9f8e292fec5eaada956565d8? I think keeping the test of `matplotlib` under the `matplotlib/` directory is more readable. (Or are we going to delete all the files under `matplotlib/` by the unification of visualization tests?)

Reply:
This is not of high priority and I think we can discuss this in another issue.